src/main/kotlin/sliv/tool/importer/view/ChoosingDirectoryView.kt

```diff
@@ -19,45 +20,71 @@ class ChoosingDirectoryView : Fragment() {
-                action {
-                    val dir = directoryChooser.showDialog(currentStage)
-                    controller.directoryPath.set(dir.absolutePath)
+        top {
+            borderpane {
+                left {
+                    vbox {
+                        label("Project directory")
+                        label {
+                            bind(controller.directoryPath)
+                        }
+                    }
+                }
+                right {
+                    button("Change") {
+                        action {
+                            val dir = directoryChooser.showDialog(currentStage)
+                            controller.directoryPath.set(dir.absolutePath)
+                        }
+                    }
                 }
             }
         }
-
         path.onChange {
-            project.set(parseDirectory(it.toString()))
-            rootTree = ProjectParser.createTreeWithFiles(project.value, rootTree)
+            if (!it.isNullOrEmpty()) {
+                project.set(parseDirectory(it.toString()))
+                directoryChooser.initialDirectory = File(path.value)
+                rootTree.children.remove(0, rootTree.children.size)
+                rootTree = ProjectParser.createTreeWithFiles(project.value, rootTree)
+            }
         }
-
-        treeview(rootTree) {
-            visibleWhen { path.isNotEmpty }
+        center {
+            padding = Insets(5.0, 0.0, 0.0, 0.0)
+            treeview(rootTree) {
+                visibleWhen { path.isNotEmpty }
+            }
         }
-
-        vbox {
-            padding = Insets(0.0, 0.0, 0.0, 10.0)
-            button("Import") {
-                isDisable = true
-                project.onChange {
-                    if (project.value.frames.isNotEmpty()) {
-                        isDisable = false
+        bottom {
+            borderpane {
+                padding = Insets(10.0, 0.0, 0.0, 0.0)
+                left {
+                    button("Cancel") {
+                        action {
+                            controller.directoryPath.set(null)
+                            MenuBarView().importer.close()
+                        }
+                        prefWidth = 180.0
                     }
                 }
-                action {
-                    val projectVal = project.value
-                    mainController.sceneFacade.visualize(projectVal.layers, projectVal.frames)
-                    MenuBarView().importer.close()
+                right {
+                    button("Import") {
+                        prefWidth = 180.0
+                        isDisable = true
+                        project.onChange {
+                            if (project.value.frames.isNotEmpty()) {
+                                isDisable = false
+                            }
+                        }
+                        action {
+                            val projectVal = project.value
```

Comment:
you use already parsed project here, but what if it will have 1 million files? when will slow moment be?